processor/src/trace/utils.rs

```diff
@@ -237,6 +237,96 @@ impl HintCycle for u64 {
     }
 }
 
+// TRACE LENGTH SUMMARY
+// ================================================================================================
+
+/// Contains the data about lengths of the trace parts.
+///
+/// - `main_trace_len` contains the length of the main trace.
+/// - `range_table_len` contains the length of the range checker table.
+/// - `chiplet_lengths` contains the trace lengths of the all chiplets (hash, bitwise, memory,
+/// kernel ROM)
+pub struct TraceLenSummary {
+    main_trace_len: usize,
+    range_table_len: usize,
+    chiplets_lengths: ChipletsLengths,
+}
+
+impl TraceLenSummary {
+    pub fn new(
+        main_trace_len: usize,
+        range_table_len: usize,
+        chiplets_lengths: ChipletsLengths,
+    ) -> Self {
+        TraceLenSummary {
+            main_trace_len,
+            range_table_len,
+            chiplets_lengths,
+        }
+    }
+
+    /// Returns length of the main trace
+    pub fn main_trace_len(&self) -> usize {
+        self.main_trace_len
+    }
+
+    /// Returns length of the range table
+    pub fn range_table_len(&self) -> usize {
+        self.range_table_len
+    }
+
+    /// Returns [ChipletsLengths] which contains trace lengths of all chilplets.
+    pub fn chiplets_lengths(&self) -> ChipletsLengths {
+        self.chiplets_lengths
+    }
```

Comment:
I think we should still have `trace_len()` and `padded_trace_len()` methods - we'd just compute them based on the data already stored here.

Also, is there a reason we use `length` for chiplets but `len` for the other two? I'd probably use the same approach. Maybe to keep things consistent we should use the following names:

- `trace_len` - this is the max of all component lengths.
- `padded_trace_len` - this is `trace_len` rounded up to the next power of two.
- `main_trace_len`
- `range_trace_len`
- `chiplets_trace_len`